docs/zh/blog/envd.md

```diff
@@ -0,0 +1,173 @@
+- 为什么我的训练到现在还没有调度起来？
+- 为什么我只是 `docker commit` 了一下提交了一个不到 10MB 的依赖，你一下把我的镜像搞大了 2GB？
+
+于是我们就此陷入了对各自遇到的稀奇古怪的环境问题的吐槽中去了。当然我们也交流了 VSCode Remote 和 Jupyter notebook 哪个更好用之类的。2022 年了，我们讨论的话题仍旧离不开程序员三大件：编程语言、编程环境、和编辑器。
+
+讨论结束后，我们一起脑洞了一个好的体验应该是什么样子的。这就是 [envd](https://github.com/tensorchord/envd) 的由来。
+
+## 问题在哪里
+
+我们认为，目前体验糟糕最主要的原因，在于目前 AI 基础设施领域的产品或项目，都是面向基础设施的（Infrastructure oriented）。它们更像是为基础设施工程师设计的，而不是为终端用户：算法工程师和数据科学家们设计的。所以这些产品都以资源利用率、GPU 共享、完善的运维支持等为卖点。
+
+但，我们作为基础设施工程师和算法工程师在技术背景和使用习惯等方面都有着显著且不可忽视的差异。这些差异都使得算法工程师们并不能很好地使用那些原本就是面向基础设施工程师设计的产品，比如 Kubernetes。
+
+当然，我们也坚定地认为不应该让算法工程师们学习如何使用 Kubernetes。Kubernetes 能够做绝大多数你想做的事情，你甚至还能[利用 Kubernetes 管理喷气式飞机](https://www.youtube.com/watch?v=YjZ4AZ7hRM0&ab_channel=CNCF%5BCloudNativeComputingFoundation%5D)。但是它就像是 Cloud 时代的 Linux kernel，没有道理让每一个工程师都了解内核的工作原理吧。
+
+如果把问题聚焦在 AI/ML 领域，那么更进一步，Docker 也可以被更简单的抽象。我们相信也没有算法工程师想了解 containerd、runc、buildkit、OCI spec 这些底层晦涩的知识。
+
+## envd 是如何设计的
+
+所以，[envd](https://github.com/tensorchord/envd) 希望能够将这些目前已经在工业界广泛使用的基础设施，面向 AI/ML 领域进一步抽象，为算法工程师和数据科学家团队们提供更加简单易用的工具和产品。[envd](https://github.com/tensorchord/envd) 的愿景是使用者再也不需要关心基础设施，只需要通过 TensorFlow、PyTorch、Jax 等框架开发模型即可。
+
+<div align="center">
+<img src='https://user-images.githubusercontent.com/5100735/188788542-269d1049-6b19-4c9d-82c2-5fb828ebdc6d.png' width='60%'>
+</div>
+
+为此，[envd](https://github.com/tensorchord/envd) 引入了基于 python 的构建文件 `build.envd`。它不再像 Kubernetes 的 YAML 配置或者是 Dockerfile 一样，围绕基础设施或者是通用容器的构建来定义配置。而是围绕 AI/ML 的模型开发与推理来进行。
+
+下面是一个非常简单的示例。展示了如何构建一个 TensorFlow 的开发环境。
+
+![](https://user-images.githubusercontent.com/5100735/188850235-633cbef7-3522-4c56-81e8-2486fd889f5b.svg)
+
+为了能够在 Kubernetes、远端服务器等不同的环境下使用 [envd](https://github.com/tensorchord/envd) 构建的环境，我们维护了一个非常轻量级的 sshd 的实现，并内置于其中。因此用户可以通过 ssh 协议连接到环境进行开发和调试。当然，你可以可以通过 [envd](https://github.com/tensorchord/envd) 提供的语法在环境中安装需要的 vscode extension，或者是配置使用 Jupyter。
+
+```python
+def build():
+    base(os="ubuntu20.04", language="python3")
+    install.vscode_extensions([
+        "ms-python.python",
+    ])
+    # Configure jupyter notebooks.
+    config.jupyter()
+    # Configure zsh.
+    shell("zsh")
+```
+
+## envd 的构建速度
+
+因为聚焦在 AI/ML 领域，因此我们对 Docker 和 buildkit 的使用进行了特殊的优化。使得在这一场景下的 [envd](https://github.com/tensorchord/envd) 构建速度比 Docker **快 6 倍**。
+
+这得益于 [envd](https://github.com/tensorchord/envd) 在各个层次上的 cache。举个例子来说明，在 Docker 中如果 Dockerfile 前面的命令缓存失效了，那么后续的命令都要重新执行，也包括 `pip install` 命令。它需要重新下载。
+
+而 [envd](https://github.com/tensorchord/envd) 会在多次构建间维护 pip index 的 cache，使得后续的构建不需要再重新下载 wheel，只需要使用已经被缓存的包即可。
+
+<table>
+<tr>
+<td> envd </td> <td> Docker </td>
+</tr>
+<tr>
+<td>
+
+```diff
+$ envd build
+=> pip install tensorflow       5s
++ => Using cached tensorflow-...-.whl (511.7 MB)
+```
+
+</td>
+<td>
+
+```diff
+$ docker build
+=> pip install tensorflow      278s
+- => Downloading tensorflow-...-.whl (511.7 MB)
+```
+
+</td>
+</tr>
+</table>
+
+在多种不同的优化加持下，[envd](https://github.com/tensorchord/envd) 的构建速度在特定场景下比 Docker 快 6 倍。
```

Comment:
I found 3 "快 6 倍" but this one is not emphasized. Not sure if you forget or if this line is redundant since it appears at the beginning of this section.